### PR TITLE
feat: add existing plugin

### PR DESCRIPTION
**Pull request by yuqizhou77**
Add an existing plugin during scaffolding.
image
image
image

Also when adding plugin, try not to modify the content of the copied plugin manifest file. Update the spec url in the new plugin manifest only when the expected destination path of API spec exists or does not within the parent folder of plugin manifest.

UI update on VSCode for selectFile question:
- Added "innerStep" and "totalInnerStep" to indicate the step of question within the group in VSCode UI
- Added "DefaultFolder" so that file selector window will be launched with expected folder

### CODE REVIEW

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -395,6 +398,12 @@
   "core.createProjectQuestion.noPlugin.detail": "Create declarative Copilot only",
   "core.createProjectQuestion.addPlugin.label": "Add plugin",
   "core.createProjectQuestion.addPlugin.detail": "Create declarative Copilot with API plugin",
+  "core.createProjectQuestion.addExistingPlugin.title": "Import from Existing - Upload files",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:401]: Is this - Import from existing uploaded files? Or 'Upload files' is a CTA?

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -395,6 +398,12 @@
...
+  "core.createProjectQuestion.addExistingPlugin.pluginManifest.placeholder": "Select your plugin manifest file",
+  "core.createProjectQuestion.addExistingPlugin.openApiSpec.placeholder": "Select the OpenAPI description used for your API plugin",
+  "core.createProjectQuestion.addPlugin.MissingRequiredProperty": "Invalid plugin manifest. Missing \"%s\"",
+  "core.createProjectQuestion.addPlugin.pluginManifestMissingApiSpec": "Invalid plugin manifest. Make sure the manifest has runtime of \"%s\" type and reference a valid API description document",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:405]: Invalid plugin manifest. Ensure the manifest has a runtime of "%s" and references a valid API description document.

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -395,6 +398,12 @@
...
+  "core.createProjectQuestion.addPlugin.pluginManifestMultipleApiSpec": "Found multiple OpenAPI description document: \"%s\".",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:406]: Found multiple Open API description documents: \"%s\"."

File: packages/fx-core/resource/package.nls.json [hunk 4]
```diff
@@ -395,6 +398,12 @@
...
   "core.aiAssistantBotOption.label": "AI Agent Bot",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:407]: AI Assistant Bot

File: packages/fx-core/resource/package.nls.json [hunk 5]
```diff
@@ -395,6 +398,12 @@
...
   "core.aiAssistantBotOption.detail": "A custom AI Agent bot in Teams using Teams AI library and OpenAI Assistants API",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:408]: Change 'AI Agent' to 'AI Assistant' in all the strings.

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:401]: I feel it is more for Upload files. It means for importing from an existing plugin.

File: packages/fx-core/resource/package.nls.json (hunk 5 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:408]: seems that it is out of the scope of this PR. Will file a bug to let PM confirm

**Review by tecton — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 6]
```diff
@@ -363,6 +364,8 @@
   "core.createProjectQuestion.llmService.azureOpenAIDeploymentName.title": "Azure OpenAI Deployment Name",
   "core.createProjectQuestion.llmService.azureOpenAIEndpoint.placeholder": "Input Azure OpenAI service endpoint now or set it later in the project",
   "core.createProjectQuestion.llmService.azureOpenAIDeploymentName.placeholder": "Input Azure OpenAI deployment name now or set it later in the project",
+  "core.createProjectQuestion.apiPlugin.importPlugin.label": "Import from existing plugin",
+  "core.createProjectQuestion.apiPlugin.importPlugin.detail": "Import from an existing API plugin file",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:368]: Import from existing API plugin file

File: packages/fx-core/resource/package.nls.json [hunk 7]
```diff
@@ -395,6 +398,12 @@
   "core.createProjectQuestion.noPlugin.detail": "Create declarative Copilot only",
   "core.createProjectQuestion.addPlugin.label": "Add plugin",
   "core.createProjectQuestion.addPlugin.detail": "Create declarative Copilot with API plugin",
+  "core.createProjectQuestion.addExistingPlugin.title": "Upload Files",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:401]: Upload files

Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:401]: I think we usually do capitalize in title, such as new project :)
image

**Review by jayzhang — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json (hunk 7 above)
Comment (supkasar) [reply packages/fx-core/resource/package.nls.json:401]: Yes, but that is incorrect. As per MS style guide, we use sentence case.

File: packages/fx-core/resource/package.nls.json [hunk 8]
```diff
@@ -395,6 +398,12 @@
...
+  "core.createProjectQuestion.addExistingPlugin.pluginManifest.placeholder": "Select your plugin manifest file",
+  "core.createProjectQuestion.addExistingPlugin.openApiSpec.placeholder": "Select the OpenAPI description used for your API plugin",
```
[the remark this answers is not in the document]
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:403]: Updated to "Select OpenAPI description document used for your API plugin" as OpenAPI should be one word,

**Review by tecton — APPROVED** [review]

**Review by supkasar — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by Alive-Fish — APPROVED** [review]

**Pull request merged**